### PR TITLE
fix: Hung up on slow rendering arts

### LINKED ISSUE #9719: Hung up on slow rendering arts
Got hung up on slow rendering arts (as I expected), minor for now
https://deploy-preview-9711--koda-canary.netlify.app/ahp/drops/sailwings

Probably I don't know how to distinguish slow and fast art renders, we should do some design where it would start doing rendering one by one. 
Even I prefer personally blast render all of them at once, at slow art types, it might take while or goes wrong as on screenshots 

Whatever works, faster safer, better for UI

image
image

_Originally posted by @yangwao in https://github.com/kodadot/nft-gallery/issues/9711#issuecomment-1991573884_

### PR DESCRIPTION
**Pull request by hassnian**
## PR Type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

## Context

>maybe if it would not be all at once?

added a max of 3 at once 

- [x] Closes #9719

## Needs QA check

- @kodadot/qa-guild please review

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=16faLfsywwNATaEfbH2ah75dn6ZmctQWpMS5G4KFhbmj5hnD)

## Screenshot 📸

- [ ] My fix has changed **something** on UI;

### CODE REVIEW

File: components/collection/drop/modal/paid/MintOverview.vue [hunk 1]
```diff
@@ -13,13 +13,15 @@
...
+              :width="46"
+              :height="46"
               :rounded="false"
               no-margin />
 
             <BaseMediaItem
+              v-if="toMintNft.canRender"
               :src="sanitizeIpfsUrl(toMintNft.image)"
               :alt="toMintNft.name"
               mime-type="text/html"
```
Comment (Jarsen136) [inline components/collection/drop/modal/paid/MintOverview.vue:27]: Do we have a placeholder image if not rendering arts?

File: composables/drop/massmint/useDropMassMint.ts [hunk 2]
```diff
@@ -18,6 +18,7 @@ export type MassMintNFT = Omit<ToMintNft, 'priceUSD'> & {
   hash: string
   sn?: number
   entropyRange: EntropyRange
+  canRender: boolean
```
Comment (preschian) [inline composables/drop/massmint/useDropMassMint.ts:21]: Is there any simpler way than this?
Is it possible to render a static image if multiple amounts? Else, render canvas

File: components/collection/drop/modal/paid/MintOverview.vue (hunk 1 above)
Comment (hassnian) [reply components/collection/drop/modal/paid/MintOverview.vue:27]: a skeleton is shown if the NFT is not rendered 

CleanShot 2024-03-28 at 13 26 08@2x

Comment (Jarsen136) [reply components/collection/drop/modal/paid/MintOverview.vue:27]: Maybe the skeleton would look like it's still loading.

File: composables/drop/massmint/useDropMassMint.ts (hunk 2 above)
Comment (hassnian) [reply composables/drop/massmint/useDropMassMint.ts:21]: hmmm , correct me if I'm wrong , but as far I know we need to render the canvas at least once to be able to upload the metadata. 

we don't have access to any static image of that nft variation, unless we have them pre-generated. (iirc that was mentioned as a possibility to speed up our minting process )

Comment (preschian) [reply composables/drop/massmint/useDropMassMint.ts:21]: > hmmm , correct me if I'm wrong , but as far I know we need to render the canvas at least once to be able to upload the metadata.

aahh, I see 👍🏻
let me re-check

File: components/collection/drop/modal/paid/MintOverview.vue (hunk 1 above)
Comment (hassnian) [reply components/collection/drop/modal/paid/MintOverview.vue:27]: ui wise nothing has changed, we still show all NFTs to be minted , but the amount of canvas that are rendered on the first load is what changed

just realized maybe the naming is confusing`MAX_RENDER_AT_ONCE_AMOUNT` 

first batch of 3 starting to render 
CleanShot 2024-03-28 at 13 38 22@2x

once one is completed we start rendering another one , basically we add another canvas.

CleanShot 2024-03-28 at 13 41 17@2x


CleanShot 2024-03-28 at 13 33 55

File: composables/drop/massmint/useDropMassMint.ts (hunk 2 above)
Comment (preschian) [reply composables/drop/massmint/useDropMassMint.ts:21]: We can get base64 static image from `toMintNft.imageDataPayload?.image`. Could you check if it is still laggy if we load multiple base64 images?

Comment (hassnian) [reply composables/drop/massmint/useDropMassMint.ts:21]: checking

File: components/collection/drop/modal/paid/MintOverview.vue (hunk 1 above)
Comment (Jarsen136) [reply components/collection/drop/modal/paid/MintOverview.vue:27]: It looks good. Thanks for providing the screenshots.

File: composables/drop/massmint/useDropMassMint.ts (hunk 2 above)
Comment (hassnian) [reply composables/drop/massmint/useDropMassMint.ts:21]: added cb48a04

## the break 

 after 2-3 canvas i does hang up for a couple of  secs 

 i'm running a m1 pro ( better than the average user's computer (I would assume)) and I see my CPU usage going crazy 

I can image some user's browser just crashing/ not responding ,  , lowering the max will make for slower minting process  , not sure how do we improve this

## overall 

I do see a difference , it's hard to see on these videos

30 nfts

canary

even though they rendered faster (no max amount) it's kinda laggy and I see some nfts going blank (there is memory cap or something?)   

https://github.com/kodadot/nft-gallery/assets/44554284/35977900-5007-4b70-8742-a80381051517

this pr 

renders slower ,(max of 3 at once) but after that its pretty smooth. 

https://github.com/kodadot/nft-gallery/assets/44554284/6b6b6ed1-fc25-45e9-9258-277ffc4d82c8

**Comment by yangwao** [comment]
first got hooked like over minute on this one 

image

but second attempt went ok!

maybe parallel after render upload it to IPFS as well? can be done on separate, this is already huge improvement if someone want to mint 5-10 pcs right away

**Comment by yangwao** [comment]
lwf
Thanks!
pay 30 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $30 @ 9.55 USD/DOT ~ 3.141 $DOT
🧗 [16faLfsywwNATaEfbH2ah75dn6ZmctQWpMS5G4KFhbmj5hnD](https://polkadot.subscan.io/account/16faLfsywwNATaEfbH2ah75dn6ZmctQWpMS5G4KFhbmj5hnD)
🔗 [0x2f43575492aab3f0039b781c427590e80c53e938a31f943d77682cba1a06087f](https://polkadot.subscan.io/extrinsic/0x2f43575492aab3f0039b781c427590e80c53e938a31f943d77682cba1a06087f)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

File: composables/drop/massmint/useDropMassMint.ts (hunk 2 above)
Comment (preschian) [reply composables/drop/massmint/useDropMassMint.ts:21]: > renders slower ,(max of 3 at once) but after that its pretty smooth.

nice, is it still faster if we remove "max of 3" stuff?

Comment (preschian) [reply composables/drop/massmint/useDropMassMint.ts:21]: would be nice if we can get rid of `isRendering` stuff

**Pull request merged**